.gitlab-ci.yml

```diff
@@ -1,86 +1,101 @@
 stages:
-  - info
+  - pre
   - lint
   - test
+  - ui-test
 
-ENV info:
-  stage: info
+variables:
+  MAIN_BRANCH: "master"
+  DEVELOP_BRANCH: "develop"
+
+default:
   tags:
-    - mac-ventura-preview
-  allow_failure: true # do not block GH PRs
+    - macos:sonoma
+    - specific:true
+
+# ┌───────────────┐
+# │ Utility jobs: │
+# └───────────────┘
+
+.run:always-on-default-branches:
+  rules:
+    - if: '$CI_COMMIT_BRANCH == $MAIN_BRANCH || $CI_COMMIT_BRANCH == $DEVELOP_BRANCH'
+      when: always
+
+.run:on-sdk-changes:
+  rules:
+    - changes:
+        - "Datadog*/**/*"
+        - "IntegrationTests/**/*"
+        - "TestUtilities/**/*"
+        - "*.yml"
```

Comment:
blocking/ This should use [`rules:changes:compare_to`](https://docs.gitlab.com/ee/ci/yaml/index.html#ruleschangescompare_to) and use diff against PR target branch. Not sure if possible with our GitLab setup. I'll have a look 👀.

Blocking because this will trick the jobs listed in GH checks. Currently it is using diff compared to previous commit on the branch, which is wrong.

Reply:
Fixed 👌. Tested 3 scenarios, all looks good:

<img width="903" alt="Screenshot 2024-06-19 at 16 38 22" src="https://github.com/DataDog/dd-sdk-ios/assets/2358722/d895c3df-871a-4208-ab38-b828fffd09de">